dbus.c

```diff
@@ -170,91 +160,69 @@ static zend_object_handlers dbus_object_handlers_dbus_array, dbus_object_handler
 static zend_object_handlers dbus_object_handlers_dbus_variant, dbus_object_handlers_dbus_set;
 static zend_object_handlers dbus_object_handlers_dbus_struct, dbus_object_handlers_dbus_object_path;
 
-typedef struct _php_dbus_obj php_dbus_obj;
-typedef struct _php_dbus_object_obj php_dbus_object_obj;
-typedef struct _php_dbus_signal_obj php_dbus_signal_obj;
-typedef struct _php_dbus_array_obj php_dbus_array_obj;
-typedef struct _php_dbus_dict_obj php_dbus_dict_obj;
-typedef struct _php_dbus_variant_obj php_dbus_variant_obj;
-typedef struct _php_dbus_set_obj php_dbus_set_obj;
-typedef struct _php_dbus_struct_obj php_dbus_struct_obj;
-typedef struct _php_dbus_object_path_obj php_dbus_object_path_obj;
-typedef struct _php_dbus_data_array php_dbus_data_array;
-
-struct _php_dbus_obj {
-	zend_object     std;
+#define PHP_DBUS_SIGNAL_IN  1
+#define PHP_DBUS_SIGNAL_OUT 2
+
+DBUS_ZEND_OBJ_STRUCT_DECL_BEGIN(php_dbus_obj)
 	DBusConnection *con;
 	int             useIntrospection;
 	HashTable       objects; /* A hash with all the registered objects that can be called */
-};
+DBUS_ZEND_OBJ_STRUCT_DECL_END();
 
-struct _php_dbus_object_obj {
-	zend_object      std;
+DBUS_ZEND_OBJ_STRUCT_DECL_BEGIN(php_dbus_object_obj)
 	php_dbus_obj    *dbus;
 	char            *destination;
 	char            *path;
 	char            *interface;
 	xmlDocPtr        introspect_xml_doc;
     xmlNode         *introspect_xml;
-};
-
-#define PHP_DBUS_SIGNAL_IN  1
-#define PHP_DBUS_SIGNAL_OUT 2
+DBUS_ZEND_OBJ_STRUCT_DECL_END();
 
-struct _php_dbus_signal_obj {
-	zend_object      std;
+DBUS_ZEND_OBJ_STRUCT_DECL_BEGIN(php_dbus_signal_obj)
```

Comment:
The use of this macro, and many others, is probably no longer useful. I suppose they were introduced to make the same code work for both PHP 5 and PHP 7.

I think I would prefer it if these macros were not added and used now, but I don't mind it too much either for more complicated macros. One liners, really ought not to be macros. (Like this, and the ``REGISTER_CLASS`` ones, for example).

Reply:
I wanted to keep git history and autorship from the old PR as-is. there is a commit that reverts obviously unneeded compat typedefs now. I would also like to get rid of some macros as well, but we can do later if that's okay for you.